app/javascript/components/users/SignInPage.spec.tsx

```diff
@@ -68,22 +68,31 @@ async function locationAssign(input: (locationAssignSpy: jest.SpyInstance<void,
 describe('Links', () => {
 	it('forgot password Link goes to correct path', async () => {
 		expect.hasAssertions();
-		locationAssign(async (locationAssignSpy: jest.SpyInstance<void, [url: string]>) => {
+		await locationAssign(async (locationAssignSpy: jest.SpyInstance<void, [url: string]>) => {
```

Comment:
This was what was causing inconsistent behavior on the tests. The problem is that locationAssign returns a promise but the test wasn't awaiting on that promise. That means the test was finishing and cleanup of the test started but the promise inside the test was still running.